src/hooks/Recoil_Hooks.js

```diff
@@ -232,6 +233,7 @@ function useInterface(): RecoilInterface {
     function useRecoilState<T>(
       recoilState: RecoilState<T>,
     ): [T, SetterOrUpdater<T>] {
+      if (!isRecoilValue(recoilState)) throw new Error('Invalid atom provided');
```

Comment:
nodes may be either atoms or selectors.  the type is `RecoilState<T>`

Reply:
yes, isRecoilValue should be checking for RecoilState.
Maybe the error message should be? `No atom or selector provided`